tests/unit/settled-test.js

```diff
@@ -281,6 +316,38 @@ module('settled', function(hooks) {
       assert.strictEqual(isSettled(), true, 'post cond');
     });
 
+    test('when router is transitioning', function(assert) {
+      assert.expect(3);
+      assert.strictEqual(isSettled(), true);
+
+      setContext(this);
+      this.owner = {
```

Comment:
Same as above here. I'd really like tests that reflect a users reality (instead of mocking).